File: src/main/java/org/carlosalcina/java/escuela/Estudiante.java
package org.carlosalcina.java.escuela;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class Estudiante {
    public String nombre;
    private List<Curso> cursos = new ArrayList<>();
    private Map<Curso, Double> calificaciones = new HashMap<>();

    public Estudiante(String nombre) {
        this.nombre = nombre;
    }

    public void asignarCurso(Curso curso) {
        cursos.add(curso);
    }

    public void asignarCalificacion(Curso curso, double calificacion) {
        if (cursos.contains(curso)) {
            calificaciones.put(curso, calificacion);
        }
    }

    public double calcularPromedio() {
        if (calificaciones.isEmpty()) return 0.0;
        return calificaciones.values().stream().mapToDouble(Double::doubleValue).average().orElse(0.0);
    }

    public void mostrarPromedio() {
        System.out.println("Promedio de " + nombre + ": " + calcularPromedio());
    }
}

File: src/main/java/org/carlosalcina/kotlin/escuela/Estudiante.kt
package org.carlosalcina.kotlin.escuela

class Estudiante(val nombre: String) {
    private val cursos = mutableListOf<Curso>()
    private val calificaciones = mutableMapOf<Curso, Double>()

    fun asignarCurso(curso: Curso) {
        cursos.add(curso)
    }

    fun asignarCalificacion(curso: Curso, calificacion: Double) {
        if (curso in cursos) {
            calificaciones[curso] = calificacion
        }
    }

    fun calcularPromedio(): Double {
        return if (calificaciones.isNotEmpty()) {
            calificaciones.values.average()
        } else 0.0
    }

    fun mostrarPromedio() {
        println("Promedio de $nombre: ${calcularPromedio()}")
    }
}